### PR TITLE
Renamed Objectives and Key Results to Outcomes and Metrics

### LINKED ISSUE #2584: Rename Objectives and Key Results to Outcomes and Metrics
Based on #2446

1) Change “Objectives and Key Results" header to “Outcomes and Metrics” 
2) Change main help text to
”Please describe how you will achieve the stated purpose of this activity in terms of Outcomes and Metrics.

An Outcome is a discrete and measurable improvement to Medicaid program management, monitoring, or administration, resulting from the delivery of IT system functionality. Outcomes should support the priorities of the Medicaid program and be directly enabled by the state’s IT project.

A Metric is a measure that would demonstrate whether a system is meeting an outcome. Metrics support both CMS-required and state-proposed outcomes, and are comprised of underlying data. In accordance with 42 CFR §433.112.”  

3) Change example text to:
“Example:
Outcome: Develop and implement an MES Module to coordinate and improve public health and environmental services for children exposed to lead.
Metrics:  1. Percentage of children in Medicaid that are receiving lead screening at 12 and 24 months of age.
2. Percentage of children in Medicaid that receive timely public health case management and environmental services based on the Childhood Lead Poisoning Prevention Protocol for lead-exposed children.
3. Percentage of children in Medicaid with lead poisoning that had timely follow-up testing by providers.”

4) Add an additional separator after example and before the Outcome/Metrics summary
5) Change Edit/Remove on summary to Edit | Delete, matching the Activity List
6) Change button to “Add another outcome”
7) Remove Target and Baseline from entry fields and summary
8) Change help text on Outcome field to “Describe a discrete and measurable improvement for this system.” 
 9) Change “key results” field to “Metric”
10) Change help text on Metrics field to “Describe a measure that would demonstrate whether this system is meeting this outcome.”
11) Allow for multiple metrics
12) Allow for deletion of metrics
13) Change button to “Add another metric”

### This task is done when...
 - [x] header has been changed
 - [x] main text has been changed
 - [x] example text has been changed
 - [x] separator added between example and Outcome/Metrics summary
 - [x] Edit/Remove button on summary has been changed to Edit | Delete
 - [x] Button text has been changed to "Add another outcome"
 - [x] Target and Baseline has been removed from entry fields and summary
 - [x] changed help text on Outcome field
 - [x] changed "key results" field to "Metric"
 - [x] changed help text on Metrics field
 - [x] allows multiple metrics
 - [x] allows deletion of metrics
 - [x] changed button to "Add another metric"

### PR DESCRIPTION
**Pull request by thetif**
Updated Objectives and Key Results to be Outcomes and Metrics.

resolves #2584 

### This pull request changes...

- Change “Objectives and Key Results" header to “Outcomes and Metrics” 
- Change main help text
- Change example text
- Add an additional separator after example and before the Outcome/Metrics summary
- Change Edit/Remove on summary to Edit | Delete, matching the Activity List
- Change button to “Add another outcome”
- Remove Target and Baseline from entry fields and summary
- Change help text on Outcome field to “Describe a discrete and measurable improvement for this system.”
- Change “key results” field to “Metric”
- Change help text on Metrics field to “Describe a measure that would demonstrate whether this system is meeting this outcome.”
- Allow for multiple metrics
- Allow for deletion of metrics
- Change button to “Add another metric”

### This pull request is ready to merge when...

- [x] Tests have been updated (and all tests are passing)
- [ ] This code has been reviewed by someone other than the original author
- [ ] The experience passes a basic manual accessibility audit (keyboard nav, screenreader, text scaling) OR an exemption is documented
- [x] The change has been documented
- [x] Associated OpenAPI documentation has been updated
- [x] Changelog is updated as appropriate

### This feature is done when...

- [ ] Design has approved the experience
- [ ] Product has approved the experience

### CODE REVIEW

**Comment by CSwartzHMA** [comment]
@thetif Is there a separate issue to change the Nav Bar? Currently the Nav Bar says "Outcomes and milestones" but the page is titled "Outcomes and Metrics". I think the Nav Bar should match the page title. 

I also noted that the Nav Bar, layer three headings are not capitalized like the other layers. I can't remember if this was by design or maybe this layer was overlooked.

**Comment by CSwartzHMA** [comment]
@thetif @beparticular  On the export view, I don't think we want to have the metrics on bold font, do we?

**Comment by thetif** [comment]
@CSwartzHMA @beparticular I updated the export formatting. I wanted to do the nav bar casing like the title, but yeah it wouldn't match the other nav bar items around it. We have a casing story that I will try to get to soon.

**Review by CSwartzHMA — APPROVED** [review]
Perfect! I am good with this.

File: web/src/reducers/apd.test.js [hunk 1]
```diff
@@ -846,28 +844,26 @@ describe('APD reducer', () => {
     it('should add a new activity OKR', () => {
```
Comment (radavis) [inline web/src/reducers/apd.test.js:844]: Might want to update the text describing this test.

File: web/src/i18n/locales/en/activities/outcomes.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+noDataNotice: There are no objectives and key results for this activity.
+
+description:
+  input:
+    label: Goal name
+    hint: >-
+      Examples of goals include: improved health outcomes for benficiaries, transformation of public health services,
+      enabling implementation of other initiatives, etc.
+
+objective:
```
Comment (radavis) [inline web/src/i18n/locales/en/activities/outcomes.yaml:41]: Is this content being used anywhere?

File: web/src/i18n/locales/en/activities/outcomes.yaml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+    1. Percentage of children in Medicaid that are receiving lead screening at 12 and 24 months of age.
+
+
+    2. Percentage of children in Medicaid that receive timely public health case management and environmental services based on the Childhood Lead Poisoning Prevention Protocol for lead-exposed children.
+
+
+    3. Percentage of children in Medicaid with lead poisoning that had timely follow-up testing by providers.
+
+noDataNotice: There are no objectives and key results for this activity.
```
Comment (radavis) [inline web/src/i18n/locales/en/activities/outcomes.yaml:32]: "There are no outcomes and metrics for this activity" ?

**Review by radavis — APPROVED** [review]
Just a few items to address. No show stoppers.

File: web/src/components/Review.js [hunk 4]
```diff
@@ -49,13 +49,14 @@ const Review = ({
...
-              |
+              {onEditClick && <strong>|</strong>}
               <Button
                 size="small"
                 variation="transparent"
                 onClick={onDeleteClick}
+                className="ds-u-color--error"
               >
-                Remove
+                Delete
```
Comment (tbolt) [inline web/src/components/Review.js:59]: Do we use "Delete" throughout the app? We should aim for some consistency with the language

**Review by tbolt — APPROVED** [review]

File: web/src/reducers/apd.test.js (hunk 1 above)
Comment (thetif) [reply web/src/reducers/apd.test.js:844]: Oh man, I thought I caught them all. Thanks!

File: web/src/i18n/locales/en/activities/outcomes.yaml (hunk 2 above)
Comment (thetif) [reply web/src/i18n/locales/en/activities/outcomes.yaml:41]: I actually don't think it's being used any where any more

File: web/src/components/Review.js (hunk 4 above)
Comment (thetif) [reply web/src/components/Review.js:59]: This was part of another ticket I think that requested it be changed to delete, so I think that's what we are moving to.

File: web/src/i18n/locales/en/activities/outcomes.yaml (hunk 3 above)
Comment (thetif) [reply web/src/i18n/locales/en/activities/outcomes.yaml:32]: Thanks, I'll update that too.

File: web/src/i18n/locales/en/activities/outcomes.yaml (hunk 2 above)
Comment (thetif) [reply web/src/i18n/locales/en/activities/outcomes.yaml:41]: I left it thought so that we don't lose the wording if we need it again.

**Review by beparticular — APPROVED** [review]

**Pull request merged**